bonfire/processor.py

```diff
@@ -30,23 +30,82 @@ def _process_template(*args, **kwargs):
     return processed_template
 
 
-def _remove_resource_config(items):
-    for i in items:
-        if i["kind"] != "ClowdApp":
+def _is_trusted_config(value, regex, component_params, path):
+    """
+    Check for presence of a trusted param being used for the value.
+
+    A trusted parameter is defined as:
+    1. matching the expected regex pattern
+    2. the parameter value is set in the component's deploy config
+
+    Returns True if we determine this is a trusted value
+    """
+    in_params = False
+    match = re.match(regex, value)
+    if match and match.groups()[0] in component_params:
+        in_params = True
+
+    log.debug(
+        "value '%s', regex r'%s', matches=%s, in params=%s", value, regex, bool(match), in_params
+    )
+
+    if match and in_params:
+        return True
+
+
+def _remove_untrusted_configs(data, params, path=None, current_dict=None, current_key=None):
+    """
+    Locate configurations within 'data' and remove them if not trusted.
+
+    Checks to see if any config matching a path listed in 'config.TRUSTED_PARAM_REGEX_FOR_PATH' is
+    found within 'data' dictionary and ensures the regex matches and that the parameter value is
+    set on the component's deploy config.
+    """
+    if not path:
```

Comment:
You are setting `path=None` as the default argument value, could you just set `path=""` and avoid this conditional assign?